src/lib.rs

```diff
@@ -57,11 +57,12 @@
 //! # }
 //! ```
 
-use std::fmt;
-use std::mem;
-use std::slice;
+#![cfg_attr(not(feature = "std"), no_std)]
 
-use std::error::Error as StdError;
+#[cfg(feature = "std")]
+use std::{fmt, mem, slice, error::Error as StdError};
+#[cfg(not(feature = "std"))]
+use core::{fmt, mem, slice};
```

Comment:
Couldn't this be changed to

```rust
use core::{fmt, mem, slice};
#[cfg(feature = "std")]
use std::error::Error as StdError;
```

Seems a bit cleaner to me.

Otherwise looks all good :) Can you update that? I'll do a release tomorrow then.

Reply:
Unfortunately, `core` isn't imported in `std` environment, so it will require an additional `extern crate core` line. Moreover, it should be hidden under `#[cfg(feature = "std")]` because core is already imported with `no_std`.

`extern crate core` isn't required in the 2018 edition, so I have migrated the crate: only minor changes are required. I can revert those changes if you don't like them. :upside_down_face:

Reply:
Migrating to 2018 is great, thanks. That was also on my list :) I'll take a look at the changes now